### PR TITLE
Document kubelet config format

### PR DESCRIPTION
**Pull request by tengqm**
This is a reference for kubelet config generated from kubernetes-sigs/reference-docs/genref tool.
More specifically, it is generated using the following command:

```shell
./genref -include kubelet-config
```

### CODE REVIEW

**Comment by tengqm** [comment]
Preview: https://deploy-preview-26590--kubernetes-io-master-staging.netlify.app/docs/reference/command-line-tools-reference/kubelet-config.v1beta1/

**Comment by kbhawkey** [comment]
👀

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md [hunk 1]
```diff
@@ -305,7 +305,7 @@ into `/var/lib/kubelet/config/init/kubelet` file.
 
 The init configuration is used for starting the kubelet on this specific node, providing an alternative for the kubelet drop-in file;
 such configuration will be replaced by the kubelet base configuration as described in following steps.
-See [set Kubelet parameters via a config file](/docs/tasks/administer-cluster/kubelet-config-file) for additional info.
+See [set kubelet parameters via a config file](/docs/tasks/administer-cluster/kubelet-config-file) for additional information.
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:308]: some of these changes are valid but adjacent to the introduction of the kubelet component config.

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md [hunk 2]
```diff
@@ -325,7 +328,8 @@ kubeadm relies on the kubelet to pull the control plane images and run them prop
 After the control plane is up, kubeadm completes the tasks described in following paragraphs.
 
 ### (optional) Write base kubelet configuration
-{{< feature-state for_k8s_version="v1.9" state="alpha" >}}
+
+{{< feature-state for_k8s_version="v1.11" state="beta" >}}
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:332]: the 1.9 == alpha state here is about the DynamicKubeletConfig feature gate, not the kubelet v1beta1 API.

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md [hunk 3]
```diff
@@ -512,17 +515,17 @@ Please note that:
...
   created with `kubeadm token`)
-- The temporary authentication resolve to a user member of  `system:bootstrappers:kubeadm:default-node-token` group which was granted
+- The temporary authentication resolve to a user member of `system:bootstrappers:kubeadm:default-node-token` group which was granted
   access to CSR api during the `kubeadm init` process
 - The automatic CSR approval is managed by the csrapprover controller, according with configuration done the `kubeadm init` process
 
 ### (optional) Write init kubelet configuration
 
-{{< feature-state for_k8s_version="v1.9" state="alpha" >}}
+{{< feature-state for_k8s_version="v1.11" state="beta" >}}
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:524]: [1]

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ For more details please see the [Network Plugin Requirements](/docs/concepts/ext
 
 | Protocol | Direction | Port Range | Purpose                 | Used By                   |
 |----------|-----------|------------|-------------------------|---------------------------|
-| TCP      | Inbound   | 6443*      | Kubernetes API server   | All                       |
+| TCP      | Inbound   | 6443\*      | Kubernetes API server   | All                       |
```
Comment (neolit123) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:81]: is the \ before * required here (escaping)?

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md (hunk 2 above)
Comment (tengqm) [reply content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:332]: According to https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/, `DynamicKubeletConfig` has become beta in 1.11.

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md (hunk 3 above)
Comment (tengqm) [reply content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:524]: According to https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/, `DynamicKubeletConfig` has become beta in 1.11.

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md (hunk 4 above)
Comment (tengqm) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:81]: Yes. My editor treats it as an opening for markdown emphasize `*something*` if not escaped.

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:308]: Correct. When scanning the page for links to adapt/revise, I saw some trivial defects and took the freedom to fix them. These changes are too trivial to warrant a separate PR.

File: content/en/docs/reference/setup-tools/kubeadm/implementation-details.md (hunk 2 above)
Comment (neolit123) [reply content/en/docs/reference/setup-tools/kubeadm/implementation-details.md:332]: right, i got confused.

**Review by neolit123 — COMMENTED** [review]
LGTM on the inclusion of the kubelet config reference.
the overall diff might need a more thorough review.

**Comment by sftim** [comment]
/sig node
@kubernetes/sig-node-pr-reviews would you be OK to review this?

See https://github.com/kubernetes/website/pull/26590#issuecomment-780417936 for a link to the preview.

**Comment by sftim** [comment]
One concern I have about generating documentation from code comments here is that the context might not be matched.
For example: in Go, `staticPodURLHeader` **is** 
> a map of slices with HTTP headers to use when accessing the `podURL`
> &hellip;
> Default: nil

However, if you're writing a Kubelet configuration file in your text editor, you're probably editing YAML. YAML doesn't have slice. YAML represents nil values as `` (not the empty string, just no value set in a map), `null` or `~`; the term “nil” is a Golang detail.

That might be OK so long as we can clarify in the page that we're describing the Golang implementation of a kubelet, rather than the `kubelet.config.k8s.io/v1beta1` format as you'd find it on a filesystem.

**Comment by tengqm** [comment]
@sftim That is actually a good point. The reference docs we generated today, including API reference and component references, all have some Golang imprints in them. In [kube-apiserver reference](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/), you see `--cors-allowed-origins stringSlice` and `--feature-gates mapStringBool`; in [ObjectMeta reference](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#ObjectMeta), you see `creationTimestamp (Time)` though the field is a "GoLang formatted string". I don't see a quick and clean solution for this in the near term.

**Comment by kbhawkey** [comment]
@tengqm , Should the configuration references publish with the 1.21 release (or before)?
There are separate PRs for each reference, but I'd think it makes sense to review as a group? 

Requesting review from SIG Node:
/cc @kubernetes/sig-node-pr-reviews

**Comment by tengqm** [comment]
> Should the configuration references publish with the 1.21 release (or before)?

This is for 1.20. For 1.21, we don't have new version of the package for reference generation yet.

**Comment by tengqm** [comment]
/hold
Depends on: https://github.com/kubernetes-sigs/reference-docs/pull/214

**Comment by tengqm** [comment]
/hold cancel
kubernetes-sigs/reference-docs#214 is in.

**Comment by sftim** [comment]
For clarity:
/retitle Document kubelet config format

**Comment by sftim** [comment]
/lgtm
Not doing technical review as this is generated from reviewed code

File: content/en/docs/concepts/configuration/configmap.md [hunk 5]
```diff
@@ -224,7 +224,7 @@ When a ConfigMap currently consumed in a volume is updated, projected keys are e
 The kubelet checks whether the mounted ConfigMap is fresh on every periodic sync.
 However, the kubelet uses its local cache for getting the current value of the ConfigMap.
 The type of the cache is configurable using the `ConfigMapAndSecretChangeDetectionStrategy` field in
-the [KubeletConfiguration struct](https://github.com/kubernetes/kubernetes/blob/{{< param "docsbranch" >}}/staging/src/k8s.io/kubelet/config/v1beta1/types.go).
+the [KubeletConfiguration struct](/docs/reference/config-api/kubelet-config.v1beta1/)).
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/configmap.md:227]: nit: I think there is an extra paren.

File: content/en/docs/reference/_index.md [hunk 6]
```diff
@@ -66,6 +68,10 @@ client libraries:
 
 * [kube-proxy configuration (v1alpha1)](/docs/reference/config-api/kube-proxy-config.v1alpha1/)
 
+## Config APIs
+
+* [kubelet config (v1beta1)](/docs/reference/config-api/kubelet-config.v1beta1/)
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:73]: nit: Should line 69 (kube-proxy configuration) appear under the "Config APIs" heading?

**Comment by kbhawkey** [comment]
Thanks @tengqm .
/approve

File: content/en/docs/reference/_index.md (hunk 6 above)
Comment (tengqm) [reply content/en/docs/reference/_index.md:73]: This subsection is a little messy when we are merging in the various config APIs. I'm cleaning and rebasing along the way. When the last piece is in, we need to go through this subsection and fix whatever nits we have here.

File: content/en/docs/concepts/configuration/configmap.md (hunk 5 above)
Comment (tengqm) [reply content/en/docs/concepts/configuration/configmap.md:227]: Eagle eye!!! Will fix.

**Pull request merged**